### PR TITLE
[2201.3.x] Add new BBE for function pointers

### PR DESCRIPTION
**Pull request by chiranSachintha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/function-pointers/function_pointers.bal [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    return func();
+}
+
+// Here, the function pointer without default values for function pointer parameters is used as a parameter. 
+function operation(function (int, int) returns int func, int v1, int v2) returns int {
+    return func(v1, v2);
+}
+
+public function main() {
+    // The function names `add` and `multiply` serves as a function pointer argument in the
```
Comment (praneesha) [inline examples/function-pointers/function_pointers.bal:25]:
```suggestion
    // The `add` and `multiply` function names serve as a function pointer argument in the
```

File: examples/function-pointers/function_pointers.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Function pointers
+
+Ballerina allows you to define variables(function pointers) of function types. The name of the function variable serves as a reference to that function when it is used in an expression context. A function pointer can be invoked similarly to how a normal function is invoked.
```
Comment (praneesha) [inline examples/function-pointers/function_pointers.md:3]:
```suggestion
Ballerina allows you to define variables (function pointers) of function types. The name of the function variable serves as a reference to that function when it is used in an expression context. A function pointer can be invoked similarly to how a normal function is invoked.
```

**Review by praneesha — APPROVED** [review]

File: examples/function-pointers/function_pointers.bal [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+int num1 = 10;
+int num2 = 100;
+
+// Here, the function pointer with default values for function pointer parameters is used as a parameter. 
+function process(function (int a = num1, int b = num2) returns int func) returns int {
+    return func();
+}
+
+// Here, the function pointer without default values for function pointer parameters is used as a parameter. 
+function operation(function (int, int) returns int func, int v1, int v2) returns int {
```
Comment (gimantha) [inline examples/function-pointers/function_pointers.bal:20]: Shall we use a verb for the function name here?

Comment (gimantha) [reply examples/function-pointers/function_pointers.bal:20]: May be `call` for this function and `callWithDefaultValues` for top one

**Review by praneesha — APPROVED** [review]

**Pull request merged**